pkg/cmd/azurekeyvaulttemplatefuncs.go

```diff
@@ -0,0 +1,97 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azsecrets"
+)
+
+const azureVaultDomain = "vault.azure.net"
+
+type azureKeyVault struct {
+	Name   string `json:"name"  mapstructure:"name"  yaml:"name"`
+	Alias  string `json:"alias" mapstructure:"alias" yaml:"alias"`
+	client *azsecrets.Client
+	cache  map[string]string
+}
+
+func (v *azureKeyVault) URL() string {
+	return "https://" + v.Name + "." + azureVaultDomain + "/"
+}
+
+type azureKeyVaultConfig struct {
+	Vaults []*azureKeyVault `json:"vaults" mapstructure:"vaults" yaml:"vaults"`
+	cred   *azidentity.DefaultAzureCredential
+}
+
+func (a *azureKeyVaultConfig) GetVaultNameFromAlias(alias string) string {
+	for _, vault := range a.Vaults {
+		if vault.Alias == alias || vault.Name == alias {
+			return vault.Name
+		}
+	}
+
+	// in case alias is not in config assume a vault name and add it to the vaults slice
+	a.Vaults = append(a.Vaults, &azureKeyVault{alias, alias, nil, nil})
+
+	return alias
+}
+
+func (a *azureKeyVaultConfig) GetSecret(secretName, vaultName string) string {
+	var err error
+
+	if a.cred == nil {
+		a.cred, err = azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	for i, vault := range a.Vaults {
+		if vault.Name == vaultName {
+			if secret, ok := a.Vaults[i].cache[secretName]; ok {
+				return secret
+			}
+
+			if vault.client == nil {
+				a.Vaults[i].client, err = azsecrets.NewClient(vault.URL(), a.cred, nil)
+
+				if err != nil {
+					panic(err)
+				}
+			}
+
+			resp, err := a.Vaults[i].client.GetSecret(context.TODO(), secretName, "", nil)
+			if err != nil {
+				panic(err)
+			}
+
+			if a.Vaults[i].cache == nil {
+				a.Vaults[i].cache = make(map[string]string)
+			}
+
+			a.Vaults[i].cache[secretName] = *resp.Value
+
+			return *resp.Value
+		}
+	}
+
+	panic(fmt.Errorf("vault %s not found", vaultName))
+}
+
+func (c *Config) azureKeyVaultTemplateFunc(args ...string) string {
+	var secretName, vaultName string
+
+	switch len(args) {
+	case 1:
+		secretName, vaultName = args[0], c.AzureKeyVault.GetVaultNameFromAlias("")
+	case 2:
+		secretName, vaultName = args[0], c.AzureKeyVault.GetVaultNameFromAlias(args[1])
+	default:
```

Comment:
This would be changed to:

```go
switch len(args) {
case 1:
  secretName, vaultName = args[0], c.AzureKeyVault.DefaultVault
case 2:
  secretName, vaultName = args[0], args[1]
default:
  panic(…)
}
```

Reply:
Implemented, but added an additional check in case no default is provided in the config.